src/luadebug/rdebug_hookmgr.cpp

```diff
@@ -463,17 +464,19 @@ struct hookmgr {
         default:
             return;
         }
-        push_callback(L);
-        luadebug::debughost::set(L, hL);
         if ((step_mask & LUA_MASKLINE) && (!stepL || stepL == hL)) {
+            push_callback(L);
+            luadebug::debughost::set(L, hL);
             luadbg_pushstring(L, "step");
             luadbg_pushinteger(L, ar->currentline);
             if (luadbg_pcall(L, 2, 0, 0) != LUADBG_OK) {
                 luadbg_pop(L, 1);
                 return;
             }
         }
-        else {
+        else if (break_mask & LUA_MASKLINE) {
```

Comment:
Possibly replace `if (break_mask & LUA_MASKLINE)` replace it with
```C
#ifdef LUAJIT_VERSION 
if (break_mask & LUA_MASKLINE) 
#endif
```

Reply:
@fesily